packages/flutter/lib/src/material/outline_button.dart

```diff
@@ -339,37 +356,46 @@ class _OutlineButtonState extends State<OutlineButton> with SingleTickerProvider
 
   // TODO(hmuller): this method is the same as FlatButton
   Color _getSplashColor(ThemeData theme, ButtonThemeData buttonTheme) {
-    if (widget.splashColor != null)
-      return widget.splashColor;
+    if (widget.splashColor != null) return widget.splashColor;
 
     switch (_getTextTheme(buttonTheme)) {
       case ButtonTextTheme.normal:
       case ButtonTextTheme.accent:
         return theme.splashColor;
       case ButtonTextTheme.primary:
         return theme.brightness == Brightness.dark
-          ? Colors.white12
-          : theme.primaryColor.withOpacity(0.12);
+            ? Colors.white12
+            : theme.primaryColor.withOpacity(0.12);
     }
     return Colors.transparent;
   }
 
+  /// The button outline has a color and a width.
+  ///
+  /// If [this.borderSize] is provided and the style is none, then that is used.
+  ///
+  /// Else, the color and width resolve to the most specific provided value,
+  /// or the first available theme value.
   BorderSide _getOutline(ThemeData theme, ButtonThemeData buttonTheme) {
-    final bool themeIsDark = theme.brightness == Brightness.dark;
-    if (widget.borderSide?.style == BorderStyle.none)
-      return widget.borderSide;
+    if (widget.borderSide?.style == BorderStyle.none) return widget.borderSide;
+
+    Color color = widget.borderSide?.color;
+    if (_pressed) {
+      color = widget.highlightedBorderColor ?? color;
+    }
+    if (!widget.enabled) {
+      color = widget.disabledBorderColor ?? color;
+    }
 
-    final Color color = widget.enabled
-      ? (_pressed
-         ? widget.highlightedBorderColor ?? theme.primaryColor
-         : (widget.borderSide?.color ??
-            (themeIsDark ? Colors.grey[600] : Colors.grey[200])))
-      : (widget.disabledBorderColor ??
-         (themeIsDark ? Colors.grey[800] : Colors.grey[100]));
+    // TODO(hmuller): should pull from a material theme default constant
+    final Color defaultColor =
+        theme.brightness == Brightness.dark ? Colors.white : Colors.black;
+    // TODO(hmuller): insert theme.surfaceColor ?? before defaultColor
```

Comment:
don't add TODOs for other people. :-)

Reply:
Reassigned back to myself, but I did discuss with hmuller because he is working on surfaceColor